### ISSUE TITLE
uncaught error when "nosub" message is received

### ISSUE THREAD

**Issue by hems** [Gregivy/simpleddp#25]
Today I noticed this error when trying to run a page which needs user login and for some weird reason it's getting the `nosub` message.

this is yielding the following error:
![image](https://user-images.githubusercontent.com/27327/79791560-6e049e80-8345-11ea-8f21-f40f5195d7df.png)

When investigating further ( i added a breakpoint on google chrome ) turns out that the "m" argument don't have an ".error" property when getting the "nosub" message:
![image](https://user-images.githubusercontent.com/27327/79791541-647b3680-8345-11ea-8fce-152a0f2d119d.png)

I believe the error comes from "await ...ready()":

https://github.com/Gregivy/simpleddp/blob/master/src/classes/ddpSubscription.js#L114

I'm running meteor 1.10.1

**Comment by Gregivy** [comment]
The `error` comes from the server, it seems the server did not throw an error but for some reason denied the subscription.

**Comment by hems** [comment]
> The `error` comes from the server, it seems the server did not throw an error but for some reason denied the subscription.

My application was doing the follow during start:

 - Try to subscribe to a logged in subscription, that subscription returned empty array []
 - User login gets resumed
 - Try to unsubscribe from said subscription ( because of the cleanup function on React.userEffect and then subscribing again logged in ).

I suspect the problem happened when I was trying to stop a subscription while logged out or concurrently subscribing and unsubscribing to the same sub.

I worked around that issue by only trying to subscribe if I have an userId.

Still, I believe since we discovered that it's possible to have a message without the `.error` I believe it would be good to check for that property and throw it only if exists?

That problem actually blocked other parts of my application that were using simpleDDP, that might have been because of the way my application works but might as well be because `simpleDDP` got in a weird state?
